projects/igniteui-angular/src/lib/core/utils.ts

```diff
@@ -451,3 +471,125 @@ export function yieldingLoop(count: number, chunkSize: number, callback: (index:
 }
 
 export function mkenum<T extends { [index: string]: U }, U extends string>(x: T) { return x; }
+
+export function reverseAnimationResolver(animation: AnimationReferenceMetadata): AnimationReferenceMetadata {
+    return oppositeAnimation.get(animation) ?? animation;
+}
+
+const oppositeAnimation: Map<AnimationReferenceMetadata, AnimationReferenceMetadata> = new Map([
```

Comment:
I feel like the animation breakup could be expanded. Currently, we don't differentiate between animations regarding the X and Y axis. In a case where we're only flipping the container horizontally, but the animation is vertical, it would still be "flipped" (even though it doesn't need to, since the overlay would still appear, for example, on the bottom of the target)

Reply:
Also, some animations are both vertical and horizonal.
For example, to display a container to the `VerticalAlignment.Top` and `HorizontalAlignment.Left` side of a component, you'd use `scaleInBr`.
Depending on the viewport, the autoposition could force several different display flips:
`HorizontalAlignment` changes to `Left` - animation should change to `scaleInBl`.  
`VerticalAlignment` changes to `Bottom` - animation should change to `scaleInTr`.
If **both** `Horizontal` and `VerticalAlignment` change - animation should change to `scaleInTl`.